src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedObj: GroupsMap<T> = {};
+
+  items.forEach((obj: T) => {
+    const property: string = `${obj[key]}`;
```

Comment:
It's unnecessary to use template literals for converting obj[key] to a string. Use 'String(obj[key])' instead.